src/SortMsg.chpl

```diff
@@ -29,176 +30,109 @@ module SortMsg
+        for idx in DD {
+          const sliceDom = domOnAxis(d, idx, axis),
+                sliced1D = removeDegenRanks(array[sliceDom], 1),
+                sliceSorted = radixSortLSD_keys(sliced1D);
+
+          forall i in sliceDom do sorted[i] = sliceSorted[i[axis]];
         }
       }
+
+      return sorted;
     }
 
-    // https://data-apis.org/array-api/latest/API_specification/generated/array_api.searchsorted.html#array_api.searchsorted
-    @arkouda.registerND
-    proc searchSortedMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, param nd: int): MsgTuple throws {
-      param pn = Reflection.getRoutineName();
-      const x1 = msgArgs.getValueOf("x1"),
-            x2 = msgArgs.getValueOf("x2"),
-            side = msgArgs.getValueOf("side"),
-            rname = st.nextName();
 
-      var gEntX1: borrowed GenSymEntry = getGenericTypedArrayEntry(x1, st),
-          gEntX2: borrowed GenSymEntry = getGenericTypedArrayEntry(x2, st);
+    proc sort(array: [?d] ?t, alg: string, axis: int): [d] t throws 
+    where ((t != real) && (t!=int) && (t!=uint(64))) {
+      throw new Error("sort does not support type %s".format(type2str(t)));
+    }
 
-      if side != "left" && side != "right" {
-        throw getErrorWithContext(
-          msg="Unrecognized side: %s".format(side),
-          lineNumber=getLineNumber(),
-          pn,
-          moduleName=getModuleName(),
-          errorClass="NotImplementedError"
-        );
-      }
+    // https://data-apis.org/array-api/latest/API_specification/generated/array_api.searchsorted.html#array_api.searchsorted
+    @arkouda.registerCommand
+    proc searchSorted(x1: [?d1] ?t, x2: [?d2] ?t2, side: string): [d2] int throws
+    where (t == real) && (t2 == real) && (d1.rank == 1) {
 
-      // TODO: add support for Float32
-      if gEntX1.dtype != DType.Float64 || gEntX2.dtype != DType.Float64 {
-        throw getErrorWithContext(
-          msg="searchsorted only supports Float64 arrays",
-          lineNumber=getLineNumber(),
-          pn,
-          moduleName=getModuleName(),
-          errorClass="NotImplementedError"
-        );
+      if side != "left" && side != "right" {
+          throw new Error("searchSorted side must be a string with value 'left' or 'right'.");
       }
 
-      sortLogger.debug(
-        getModuleName(),pn,getLineNumber(),
-        "cmd: %s, x1: %s, x2: %s, side: %s, rname: %s, dtype: %?, nd: %i".format(
-          cmd, x1, x2, side, rname, gEntX1.dtype, nd
-        )
-      );
-
-      const e1 = toSymEntry(gEntX1, real, 1),
-            e2 = toSymEntry(gEntX2, real, nd);
-      var ret = makeDistArray((...e2.a.domain.shape), int);
+      var ret = makeDistArray((...x2.shape), int);
 
       proc doSearch(const ref a1: [] real, const ref a2: [?d] real, cmp) {
         forall idx in ret.domain {
-          const (_, i) = Search.binarySearch(a1, a2[idx], cmp);
+          const (_, i) = Search.binarySearch(x1, x2[idx], cmp);
```

Comment:
The `a1` and `a2` formals aren't being used. I think they should either be removed from `doSearch`, or used here instead of `x1` and `x2`.

Reply:
Good catch!  I made a change; let me know what you think.

Reply:
Looks good